services/meta/data.go

```diff
@@ -613,6 +613,10 @@ func (data *Data) SetPrivilege(name, database string, p influxql.Privilege) erro
 		return ErrUserNotFound
 	}
 
+	if data.Database(database) == nil {
```

Comment:
I'm pretty sure this gets called in plutonium and can now err out where previously it couldn't.  This could potentially corrupt the meta store if it replays in the raft log.